### PR TITLE
Update compiletest's dependencies

### PR DESCRIPTION
**Pull request by alexcrichton**
Drop rustc-serialize, add winapi 0.3 and Serde!

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
This seems fine to me though I do worry somewhat about switching to serde since the diagnostics are serialized by rustc-serialize. But if tests pass, then I'm happy.

@bors r+

File: src/tools/compiletest/src/json.rs [hunk 1]
```diff
@@ -9,15 +9,15 @@
...
+use serde_json;
 use std::str::FromStr;
 use std::path::Path;
 use runtest::ProcRes;
 
 // These structs are a subset of the ones found in
 // `syntax::json`.
 
-#[derive(RustcEncodable, RustcDecodable)]
+#[derive(Serialize, Deserialize)]
```
Comment (messense) [inline src/tools/compiletest/src/json.rs]: Does deriving `Serialize` really need? I did the same thing on [`compiletest-rs`](https://github.com/laumann/compiletest-rs/pull/104) crate several days ago and found that it's not needed for that crate. Not sure about this in-tree version though.

Comment (alexcrichton) [reply src/tools/compiletest/src/json.rs]: Excellent point!

**Pull request merged**